torchci/lib/bot/cancelWorkflowsOnCloseBot.ts

```diff
@@ -0,0 +1,45 @@
+import { Probot } from "probot";
+
+function cancelWorkflowsOnCloseBot(app: Probot): void {
+  app.on("pull_request.closed", async (ctx) => {
+    const owner = ctx.payload.repository.owner.login;
+    const repo = ctx.payload.repository.name;
+    const senderLogin = ctx.payload.sender.login;
+    const senderId = ctx.payload.sender.id;
+    const headSha = ctx.payload.pull_request.head.sha;
+
+    if (
+      senderLogin == "pytorchmergebot" ||
+      senderLogin == "pytorchbot" ||
+      senderLogin == "pytorch-bot[bot]" ||
+      senderId == 97764156 || // pytorchmergebot's id
+      senderId == 54816060 || // pytorch-bot's id
+      senderId == 21957446 // pytorchbot's id
+    ) {
+      return;
+    }
+
+    const workflowRuns = await ctx.octokit.actions.listWorkflowRunsForRepo({
```

Comment:
Consider wrapping this inside the pagniation API to future proof this against there being over 30 runs one day